feature/google/src/main/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -775,10 +776,12 @@ class BillingWrapper(
         val googleProduct = storeProduct.googleProduct
         if (token == null) {
             errorLog("PurchaseOption must have a token with BC5.") // TODOBC5: Improve and move error message
+            purchasesUpdatedListener?.onPurchasesFailedToUpdate(PurchasesError(PurchasesErrorCode.UnknownError))
             return null
         }
         if (googleProduct == null) {
             errorLog("Product must be a Google Product.") // TODOBC5: Improve and move error message
+            purchasesUpdatedListener?.onPurchasesFailedToUpdate(PurchasesError(PurchasesErrorCode.UnknownError))
```

Comment:
I noticed we weren't calling anything in case of these errors. This would mean the sdk would continue thinking a purchase was in progress. I'm calling the listener we call on the `onPurchasesUpdated` listener when a billing error happens. I choose `UnknownError` for now. I could add a new type of error but if we ever get here, it would mean a bug in our side so I didn't want to add a new public error for this. 

We should add the underlying error message here. I didn't do that since I know that was done in a different branch.

Reply:
great catch... i wonder if we could also error earlier, like in getOfferings if we can't make GoogleStoreProducts. something to think about

Reply:
> I think in this case, we should add the logic in the function that calls createPurchaseParams. We should return either BillingFlowParams or an error, and then decide there what to do with the error. This feels like a dangerous side effect

You're right that's much safer. Will change it!

> great catch... i wonder if we could also error earlier, like in getOfferings if we can't make GoogleStoreProducts. something to think about

We should do that yeah, but since we also have to downcast and check for the token here, we need to gracefully handle it so I think we also need to do this here. But yeah, it should never get into this state if we do it right.